lte/gateway/python/scripts/generate_oai_config.py

```diff
@@ -282,8 +282,7 @@ def _get_default_slice_differentiator_type_config(service_mconfig: object) -> st
 
     Returns: slice differentiator config value
     """
-
-    if service_mconfig.default_slice_differentiator is not None:
+    if len(service_mconfig.default_slice_differentiator):
```

Comment:
what is the reason for this change?

Reply:
With GRPC/Protobuf an empty string is "" (len = 0)... Tested with following program to confirm                                
if service_mconfig.default_slice_differentiator is not None:
    os.write(fd, str.encode(str(len(service_mconfig.default_slice_differentiator))))
    os.write(fd, str.encode(str(len(service_mconfig.default_slice_differentiator))))

Output : ** SLICE DIFFERNTIATOR IS NOT NONE ** : len -0

Hence in mme.conf the default value "" and not the Default Value return by generate_oai_config script..

I guess we need to change other string variables on similar lines.

Reply:
if these things are undefined, I think the streamed down gateway.mconfig will not have the fields at all.  @ardzoht what do you say?

Reply:
The only case where you should expect an empty string from generated mconfigs is if the cloud config specifically returns that, but that should not be the case, as the default value for string types on the cloud configs is `"string"`. And if the config is not specified (as Ulas points out), there won't be any entry on the config at all. So `if service_mconfig.default_slice_differentiator is not None` is the correct handler for this.

Reply:
My observation (DEFAULT VALUE : 0xffffff)

1. No entry for defaultSliceDifferentiator with : len(service_mconfig.default_slice_differentiator)   
-------------------------------------------------------------------------------------------------------------
vagrant@magma-dev-focal:~/magma/lte/gateway$ cat configs/gateway.mconfig | grep defaultSliceDifferentiator
vagrant@magma-dev-focal:~/magma/lte/gateway$

vagrant@magma-dev-focal:~/magma/lte/gateway$ cat ~/DUMP_SLICE
 ** SLICE DIFFERNTIATOR IS NOT NONE ** : len -0vagrant@magma-dev-focal:~/magma/lte/gateway$

vagrant@magma-dev-focal:~/magma/lte/gateway$ cat /var/opt/magma/tmp/mme.conf  | grep  DEFAULT_SLICE_DIFFERENTIATOR
       { MCC="001" ; MNC="01"; DEFAULT_SLICE_SERVICE_TYPE="1" ; DEFAULT_SLICE_DIFFERENTIATOR="0xffffff"; }   <<<<<< Default value Set as expected.
vagrant@magma-dev-focal:~/magma/lte/gateway$

2. No entry for defaultSliceDifferentiator with : if service_mconfig.default_slice_differentiator is not None: 
--------------------------------------------------------------------------------------------------------------
vagrant@magma-dev-focal:~/magma/lte/gateway$  cat configs/gateway.mconfig | grep defaultSliceDifferentiator
vagrant@magma-dev-focal:~/magma/lte/gateway$

vagrant@magma-dev-focal:~/magma/lte/gateway$ cat ~/DUMP_SLICE
 ** SLICE DIFFERNTIATOR IS NOT NONE ** : len -0vagrant@magma-dev-focal:~/magma/lte/gateway$

vagrant@magma-dev-focal:~/magma/lte/gateway$ cat /var/opt/magma/tmp/mme.conf  | grep  DEFAULT_SLICE_DIFFERENTIATOR
       { MCC="001" ; MNC="01"; DEFAULT_SLICE_SERVICE_TYPE="1" ; DEFAULT_SLICE_DIFFERENTIATOR=""; }
vagrant@magma-dev-focal:~/magma/lte/gateway$  <<<<<< Default value not set

Reply:
[Length-Experiment-Output.txt](https://github.com/magma/magma/files/7180279/Length-Experiment-Output.txt)

Attaching the output of experiment.

Reply:
@panyogesh is orc8r connected to your set up? If service_mconfig.default_slice_differentiator is None, then len will fail.

Reply:
There are a couple of things to note here:

1. Currently there is no `defaultSliceDifferentiator` entry on the gateway.mconfig, so there won't be a parsed entry on the `service_mconfig` python object
2. Checking for `service_mconfig.default_slice_differentiator` existance should be enough, as Ulas says the call on `len()` will complain if there is no entry on there, which is this exact case.

So my comment still applies, I think we should just check for the entry existance `if service_mconfig.default_slice_differentiator is not None` to handle this entry. And it would be a good idea to include an entry for this on the `gateway.mconfig` default json file.

Reply:
And just to note, currently there is no Orc8r configs to build from for these entries, if the plan is to also add the configs to the cloud, `magma/lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml` should be updated to include these configs (either on gateway or network EPC configs?), this can be done on a separate PR but just wanted to bring the awareness for this. cc @ulaskozat